deployment/internal/home_chain.go

```diff
@@ -0,0 +1,247 @@
+		mapTransmitterKey[hexutil.Encode(node.TransmitterKey)] = struct{}{}
+	}
+	return nil
+}
+
+// we can't use the EVM one because we need the 32 byte transmitter address
+type MultiOCR3BaseOCRConfigArgs struct {
+	ConfigDigest                   [32]byte
+	OcrPluginType                  uint8
+	F                              uint8
+	IsSignatureVerificationEnabled bool
+	Signers                        [][]byte
+	Transmitters                   [][]byte
+}
+
+// BuildSetOCR3ConfigArgs builds the OCR3 config arguments for the OffRamp contract
+// using the donID's OCR3 configs from the CCIPHome contract.
+func BuildSetOCR3ConfigArgs(
+	donID uint32,
+	ccipHome *ccip_home.CCIPHome,
+	destSelector uint64,
+	configType globals.ConfigType,
+) ([]MultiOCR3BaseOCRConfigArgs, error) {
+	chainCfg, err := ccipHome.GetChainConfig(nil, destSelector)
+	if err != nil {
+		return nil, fmt.Errorf("error getting chain config for chain selector %d it must be set before OCR3Config set up: %w", destSelector, err)
+	}
+	var offrampOCR3Configs []MultiOCR3BaseOCRConfigArgs
+	for _, pluginType := range []ccipocr3.PluginType{ccipocr3.PluginTypeCCIPCommit, ccipocr3.PluginTypeCCIPExec} {
+		ocrConfig, err2 := ccipHome.GetAllConfigs(&bind.CallOpts{
+			Context: context.Background(),
+		}, donID, uint8(pluginType))
+		if err2 != nil {
+			return nil, err2
+		}
+
+		configForOCR3 := ocrConfig.ActiveConfig
+		// we expect only an active config
+		switch configType {
+		case globals.ConfigTypeActive:
+			if ocrConfig.ActiveConfig.ConfigDigest == [32]byte{} {
+				return nil, fmt.Errorf("invalid OCR3 config state, expected active config, donID: %d, activeConfig: %v, candidateConfig: %v",
+					donID, hexutil.Encode(ocrConfig.ActiveConfig.ConfigDigest[:]), hexutil.Encode(ocrConfig.CandidateConfig.ConfigDigest[:]))
+			}
+		case globals.ConfigTypeCandidate:
+			if ocrConfig.CandidateConfig.ConfigDigest == [32]byte{} {
+				return nil, fmt.Errorf("invalid OCR3 config state, expected candidate config, donID: %d, activeConfig: %v, candidateConfig: %v",
+					donID, hexutil.Encode(ocrConfig.ActiveConfig.ConfigDigest[:]), hexutil.Encode(ocrConfig.CandidateConfig.ConfigDigest[:]))
+			}
+			configForOCR3 = ocrConfig.CandidateConfig
+		}
+
+		if err := validateOCR3Config(destSelector, configForOCR3.Config, &chainCfg); err != nil {
+			return nil, err
+		}
+
+		var signerAddresses [][]byte
+		var transmitterAddresses [][]byte
+		for _, node := range configForOCR3.Config.Nodes {
+			signerAddresses = append(signerAddresses, node.SignerKey)
+			transmitterAddresses = append(transmitterAddresses, node.TransmitterKey)
+		}
+
+		offrampOCR3Configs = append(offrampOCR3Configs, MultiOCR3BaseOCRConfigArgs{
+			ConfigDigest:                   configForOCR3.ConfigDigest,
+			OcrPluginType:                  uint8(pluginType),
+			F:                              configForOCR3.Config.FRoleDON,
+			IsSignatureVerificationEnabled: pluginType == ccipocr3.PluginTypeCCIPCommit,
```

Comment:
ccipocr3.PluginType was recently added by @ogtownsend to common, we should use them everywhere in core as well and retire the old types